src/main/java/com/synopsys/integration/detect/workflow/componentlocationanalysis/GenerateComponentLocationAnalysisOperation.java

```diff
@@ -0,0 +1,49 @@
+package com.synopsys.integration.detect.workflow.componentlocationanalysis;
+
+import com.synopsys.integration.blackduck.api.generated.view.DeveloperScansScanView;
+import com.synopsys.integration.detect.configuration.DetectUserFriendlyException;
+import com.synopsys.integration.detect.configuration.enumeration.ExitCodeType;
+import com.synopsys.integration.detect.workflow.bdio.BdioResult;
+import com.synopsys.integration.detect.workflow.file.DetectFileUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+public class GenerateComponentLocationAnalysisOperation {
+    private static final String CLL_INPUT_FILE_NAME = "components-source.json";
+    private static final String CLL_OUTPUT_FILE_NAME = "components-with-locations.json";
+    private final BdioToCLLComponentTransformer bdioTransformer;
+    private final ScanResultToCLLComponentTransformer scanResultTransformer;
+
+    public GenerateComponentLocationAnalysisOperation(BdioToCLLComponentTransformer bdioTransformer, ScanResultToCLLComponentTransformer scanResultTransformer) {
+        this.bdioTransformer = bdioTransformer;
+        this.scanResultTransformer = scanResultTransformer;
+    }
+
+    public static File forNonPersistentOnlinePkgMngrScan(List<DeveloperScansScanView> rapidFullResults, File scanOutputFolder) throws DetectUserFriendlyException {
+        // In Part II:
+            // transform rapid scan results -> CLL input
+            // call library w/ CLL input
+        return generatePlaceHolderJsonFileForNow(scanOutputFolder);
+    }
+
+    public static File forOfflinePkgMngrScan(BdioResult bdio, File scanOutputFolder) throws DetectUserFriendlyException {
```

Comment:
I'm assuming these are placeholder method names for now and would be updated to more descriptive names based on how they end up being implemented & what they generate?

Reply:
Renamed methods to `locateComponentsforNonPersistentOnlineDetectorScan(..)` and `locateComponentsForOfflineDetectorScan(..)`